utils/programs/asa_approve.teal

```diff
@@ -0,0 +1,531 @@
+txn NumAppArgs
```

Comment:
Can you please add some comments on what this script is doing? 
An overview similar to utils/programs/globcheck.teal  with be super helpful here.